### PR TITLE
[E2E] Update CAPA version in tests

### LINKED ISSUE #3879: [E2E] Update E2E tests to use CAPA v2.0.x
/area testing

Our E2E tests are using CAPA v1.6.x which should be updated to v 2.0.x as we have be new release in place.
We should also consider upgrading the kubernetes patch version to 1.2.4 which is currently 1.24.0, so as to use the latest AMIs rebuilt with containerd version 1.6.6.

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/area testing
/kind support

**What this PR does / why we need it**:
This PR updates the CAPA version used to v2.0.x. It also updates k8s versions used in the tests to v1.24.4.
For EKS upgrade tests, the upgrade path is updated from 1.22.4 to 1.24.4

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3879 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/suites/managed/cluster.go [hunk 1]
```diff
@@ -101,8 +101,6 @@ func ManagedClusterSpec(ctx context.Context, inputGetter func() ManagedClusterSp
 	verifySecretExists(ctx, fmt.Sprintf("%s-kubeconfig", input.ClusterName), input.Namespace.Name, bootstrapClient)
 	verifySecretExists(ctx, fmt.Sprintf("%s-user-kubeconfig", input.ClusterName), input.Namespace.Name, bootstrapClient)
 
-	time.Sleep(2 * time.Minute) //TODO: replace with an eventually on the aws-iam-auth check
```
Comment (Skarlso) [inline test/e2e/suites/managed/cluster.go:104]: Heh, neat.

File: test/e2e/data/e2e_conf.yaml [hunk 2]
```diff
@@ -84,14 +84,14 @@ providers:
   - name: aws
     type: InfrastructureProvider
     versions:
-      - name: v1.5.0 # latest published release in the v1beta1 series; this is used for v1beta1 --> v1beta2 clusterctl upgrades test only.
-        value: "https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v1.5.0/infrastructure-components.yaml"
+      - name: v1.5.2 # latest published release in the v1beta1 series; this is used for v1beta1 --> v1beta2 clusterctl upgrades test only.
```
Comment (Skarlso) [inline test/e2e/data/e2e_conf.yaml:87]: This should be `v1.5.1` shouldn't it?

Comment (Ankitasw) [reply test/e2e/data/e2e_conf.yaml:87]: We have 1.5.2 release in place, so I thought we shall test with all the changes, shouldn't we? Wdyt?

Comment (Skarlso) [reply test/e2e/data/e2e_conf.yaml:87]: Oh, never mind. I think GitHub has problems. It didn't show the tag or didn't allow me to download this file.

**Comment by Skarlso** [comment]
Can we kick the link checker?

**Comment by Ankitasw** [comment]
> Can we kick the link checker?

Not sure how to do kickstart github workflow manually 😓

File: test/e2e/data/e2e_conf.yaml (hunk 2 above)
Comment (Ankitasw) [reply test/e2e/data/e2e_conf.yaml:87]: Ok cool 🙂

**Comment by Skarlso** [comment]
Ah, yeah. And a force push would cancel the e2e test now. Eh, I guess it's fine. :D

**Comment by Ankitasw** [comment]
> Ah, yeah. And a force push would cancel the e2e test now. Eh, I guess it's fine. :D

Although it should not be an issue if it's in failed state, as we really don't have any doc/md changes in this PR.

**Comment by Skarlso** [comment]
/approve
/hold

until e2e; it looks good so far

**Comment by Ankitasw** [comment]
E2e tests passed
/unhold

**Comment by Skarlso** [comment]
eh, apparently, it doesn't merge it until that dang checker is resolved. :/

**Comment by Ankitasw** [comment]
> eh, apparently, it doesn't merge it until that dang checker is resolved. :/

I can try pushing new blank commit

**Comment by Ankitasw** [comment]
Since blank commit was pushed after lgtm tag
/lgtm

**Pull request merged**